ci/create_latest_runtime_meta.py

```diff
@@ -24,9 +40,15 @@ def main():
         if req.startswith("python "):
             requirements[idx] = f"python =={python_version}"
 
-    # File compatible with `mamba install --file <...>`
-    with open("latest.txt", "w") as f:
-        f.write("\n".join(requirements))
+    # Optionally install the development version of `dask` and `distributed`
+    if os.environ.get("TEST_UPSTREAM", False):
+        dev_req = {
+            "pip": [
+                f"git+https://github.com/dask/dask.git@{get_latest_commit('dask/dask')}",
+                f"git+https://github.com/dask/distributed.git@{get_latest_commit('dask/distributed')}",
+            ]
```

Comment:
It would be good to eventually move from `pip` here to use directly install with conda from the https://anaconda.org/dask/ channel. I tried this locally but ran into https://github.com/dask/dask/issues/8984. 

I think this is good enough for now, and we can open a ticket to switch to pure-conda once https://github.com/dask/dask/issues/8984 is resolved.

Reply:
For reference, something like 

```diff
diff --git a/ci/create_latest_runtime_meta.py b/ci/create_latest_runtime_meta.py
index 908f0e8..780ba13 100644
--- a/ci/create_latest_runtime_meta.py
+++ b/ci/create_latest_runtime_meta.py
@@ -3,24 +3,20 @@ from __future__ import annotations
 import json
 import os
 import pathlib
+import shlex
+import subprocess
 import sys
-from urllib.request import Request, urlopen
 
 import yaml
 from jinja2 import Environment, FileSystemLoader, select_autoescape
+from packaging.requirements import Requirement
 
 
-def get_latest_commit(repository):
-    org, repo = repository.split("/")
-    headers = {}
-    if os.environ.get("GITHUB_TOKEN", False):
-        headers = {"Authorization": f"token {os.environ['GITHUB_TOKEN']}"}
-    request = Request(
-        f"https://api.github.com/repos/{org}/{repo}/commits?per_page=1", headers=headers
-    )
-    response = urlopen(request).read()
-    data = json.loads(response.decode())
-    return data[0]["sha"]
+def get_latest_conda(package):
+    cmd = f"conda search --json --override-channels -c dask/label/dev {package}"
+    out = subprocess.check_output(shlex.split(cmd), text=True)
+    result = json.loads(out)[package][-1]
+    return f"{result['version']}={result['build']}"
 
 
 def main():
@@ -35,23 +31,26 @@ def main():
 
     # Ensure Python is pinned to X.Y.Z version currently being used
     python_version = ".".join(map(str, tuple(sys.version_info)[:3]))
-    assert sum([req.startswith("python ") for req in requirements]) == 1
     for idx, req in enumerate(requirements):
-        if req.startswith("python "):
+        package_name = Requirement(req).name
+        if package_name == "python":
             requirements[idx] = f"python =={python_version}"
 
+    # File compatible with `mamba env create --file <...>`
+    env = {"channels": ["conda-forge"]}
+
     # Optionally install the development version of `dask` and `distributed`
+    # from `dask/label/dev` conda channel
     if os.environ.get("TEST_UPSTREAM", False):
-        dev_req = {
-            "pip": [
-                f"git+https://github.com/dask/dask.git@{get_latest_commit('dask/dask')}",
-                f"git+https://github.com/dask/distributed.git@{get_latest_commit('dask/distributed')}",
-            ]
-        }
-        requirements.append(dev_req)
+        env["channels"].append("dask/label/dev")
+
+        upstream_packages = {"dask"}
+        # upstream_packages = {"dask", "distributed"}
+        for idx, req in enumerate(requirements):
+            package_name = Requirement(req).name
+            if package_name in upstream_packages:
+                requirements[idx] = f"{package_name} =={get_latest_conda(package_name)}"
 
-    # File compatible with `mamba env create --file <...>`
-    env = {"channels": ["conda-forge"]}
     env["dependencies"] = requirements
     with open("latest.yaml", "w") as f:
         yaml.dump(env, f)
````

is what we want to switch over to conda-only